ocaml/xapi/helpers.ml

```diff
@@ -1234,3 +1234,108 @@ let host_supports_hvm ~__context host =
 let env_with_path env_vars =
   Array.of_list @@
   List.map (fun (k,v) -> Printf.sprintf "%s=%s" k v) (("PATH", String.concat ":" Forkhelpers.default_path)::env_vars)
+
+(** wrapper around the stunnel@xapi systemd service.
+  * there exist scripts (e.g. xe-toolstack-restart) which also manipulate
+    the stunnel daemon but they do this directly (not via ocaml). *)
+module Stunnel : sig
+  (** restart stunnel, possibly changing the config file *)
+  val restart : __context:Context.t -> accept:string -> unit
+end = struct
+
+  let cert = !Xapi_globs.server_cert_path
+
+  (** protect ourselves from concurrent writes to files *)
+  module File : sig
```

Comment:
* A better name for the module might be `Config`. 
* Use `update` for `update_config`
* Use `generate_pem`. It is already idempotent - I would not encode this in the name. But you could add a `?(force=true)` to enforce an update.